pkg/query/flamegraph_arrow.go

```diff
@@ -249,10 +256,14 @@ func generateFlamegraphArrowRecord(ctx context.Context, mem memory.Allocator, tr
 		}
 	}
 
+	ctx, spanNewRecord := tracer.Start(ctx, "NewRecord")
```

Comment:
wouldn't it be cleaner to just defer end the span here instead of before both potential returns?

Reply:
I was thinking about that too. It would work since it's basically the last function call, yes. I felt like being explicit here, might be nicer.